10.detect-separation/scripts/nbconverted/3.visualize-separation.r

```diff
@@ -0,0 +1,437 @@
+
+print(best_k_dim)
+print(best_seed)
+print(best_feature)
+
+feature_file <- file.path("..",
+                          "2.sequential-compression",
+                          "results",
+                          "TCGA_results",
+                          "ensemble_z_matrices",
+                          paste0("tcga_components_", best_k_dim),
+                          paste0("model_", best_seed, "_z_matrix.tsv.gz"))
+
+top_feature_df <- readr::read_tsv(feature_file,
+                                  col_types = readr::cols(
+                                    .default = readr::col_double(),
+                                    sample_id = readr::col_character()
+                                  )
+) %>%
+  dplyr::select(sample_id, !!best_feature)
+
+head(top_feature_df)
+
+dim(top_feature_df)
+
+full_tcga_ids <- c()
+for (tcga_id in strsplit(top_feature_df$sample_id, "-")) {
+  tcga_id_unlist <- unlist(tcga_id)
+  sample_id <- paste0(tcga_id_unlist[1], "-", tcga_id_unlist[2], "-", tcga_id_unlist[3])
+  full_tcga_ids <- c(full_tcga_ids, sample_id)
+}
+
+top_feature_df$tcga_id <- full_tcga_ids
+top_feature_df <- top_feature_df %>%
+  dplyr::left_join(tcga_pheno_df, by = c("tcga_id" = "bcr_patient_barcode"))
+
+top_feature_df$gender[is.na(top_feature_df$gender)] <- "Unlabeled"
+top_feature_df$gender <- dplyr::recode_factor(top_feature_df$gender, "MALE" = "Male", "FEMALE" = "Female")
+top_feature_df$gender <- factor(top_feature_df$gender, levels = c("Female", "Male", "Unlabeled"))
+
+tcga_sex_gg <- ggplot(top_feature_df,
+                      aes(y = vae_16,
+                          x = gender,
+                          fill = gender)) +
+  geom_boxplot(alpha = 0.5,
+               outlier.alpha = 0) +
+  geom_jitter(aes(color = gender),
+              size = 0.2,
+              alpha = 0.2,
+              width = 0.3) +
+  scale_fill_manual(labels = c("Male" = "Male",
+                               "Female" = "Female",
+                               "Unlabeled" = "Unlabeled"),
+                    values = c("Male" = "#42c8f4",
+                               "Female" = "#f489c9",
+                               "Unlabeled" = "grey25")) +
+  scale_color_manual(labels = c("Male" = "Male",
+                                "Female" = "Female",
+                                "Unlabeled" = "Unlabeled"),
+                     values = c("Male" = "#42c8f4",
+                                "Female" = "#f489c9",
+                                "Unlabeled" = "grey25")) +
+  theme_bw() +
+  ylab("VAE Feature 16 (k = 20)") +
+  xlab("") +
+  ggtitle("TCGA Sex") +
+  theme(axis.title.x = element_text(size = 7),
+        axis.title.y = element_text(size = 7),
+        axis.text.x = element_text(size = 6),
+        axis.text.y = element_text(size = 6),
+        plot.title = element_text(size = 8,
+                                  hjust = 0.5),
+        legend.position = "none")
+
+tcga_sex_gg
```

Comment:
Looks like you have a very long set of warnings here -- any idea what this is?

Reply:
yeah, its b/c of the `readxl` package warning that the column expected a specific data_type.